pkg/minikube/machine/cache_images.go

```diff
@@ -596,6 +596,10 @@ func PullImages(images []string, profile *config.Profile) error {
 			err = pullImages(crMgr, images)
 			if err != nil {
 				failed = append(failed, m)
+				// Rate limit check
+				if strings.Contains(err.Error(), "429 Too Many Requests") {
```

Comment:
Using strings.Contains() means we are not using a public API. This is too fragile to depend on. Do we have an error code field or error name we can check instead?